### PR TITLE
Drep certification registration: add parameters for anchor

### LINKED ISSUE #198: drep registration certificate mixed with spo registration (clean-up)
The drep registration certificate is mixed with the stake pool registration certificate.  This needs to be cleaned up so that  `drep registration-certificate ` only has the options needed for pridcing the drep registration certificate

So we go from 

```
Usage: cardano-cli conway governance drep registration-certificate
                                                                     ( ( --stake-pool-verification-key STRING
                                                                       | --cold-verification-key-file FILE
                                                                       )
                                                                       ( --vrf-verification-key STRING
                                                                       | --vrf-verification-key-file FILE
                                                                       )
                                                                       --pool-pledge LOVELACE
                                                                       --pool-cost LOVELACE
                                                                       --pool-margin RATIONAL
                                                                       ( --pool-reward-account-verification-key STRING
                                                                       | --pool-reward-account-verification-key-file FILE
                                                                       )
                                                                       ( --pool-owner-verification-key STRING
                                                                       | --pool-owner-stake-verification-key-file FILE
                                                                       )
                                                                       [ [--pool-relay-ipv4 STRING]
                                                                         [--pool-relay-ipv6 STRING]
                                                                         --pool-relay-port INT
                                                                       | --single-host-pool-relay STRING
                                                                         [--pool-relay-port INT]
                                                                       | --multi-host-pool-relay STRING
                                                                       ]
                                                                       [--metadata-url URL
                                                                         --metadata-hash HASH]
                                                                       ( --mainnet
                                                                       | --testnet-magic NATURAL
                                                                       )
                                                                     | ( --stake-verification-key STRING
                                                                       | --stake-verification-key-file FILE
                                                                       | --stake-script-file FILE
                                                                       | --stake-address ADDRESS
                                                                       )
                                                                       --key-reg-deposit-amt NATURAL
                                                                     | ( --drep-verification-key STRING
                                                                       | --drep-verification-key-file FILE
                                                                       | --drep-key-hash HASH
                                                                       )
                                                                       --key-reg-deposit-amt NATURAL
                                                                     )
                                                                     --out-file FILE
```

to 
```
Usage: cardano-cli conway governance drep registration-certificate
                                                                       (
                                                                       | --drep-verification-key STRING
                                                                       | --drep-verification-key-file FILE
                                                                       | --drep-key-hash HASH
                                                                       )
                                                                       [
                                                                        --drep-metadata-url URL
                                                                        --drep-metadata-hash HASH
                                                                       ]
                                                                      --key-reg-deposit-amt NATURAL
                                                                     --out-file FILE
```

Roadmap:

- [x] Merge https://github.com/input-output-hk/cardano-cli/pull/309
- [x] Merge https://github.com/input-output-hk/cardano-api/pull/284
- [x] Release cardano-api
- [x] Merge https://github.com/input-output-hk/cardano-cli/pull/324
- [ ] Merge https://github.com/input-output-hk/cardano-cli/pull/330

### PR DESCRIPTION
**Pull request by smelc**
# Changelog

```yaml
- description: |
    Add `--drep-metadata-url` and `--drep-metadata-hash` to the drep registration certificate command
# uncomment types applicable to the change:
  type:
  - feature        # introduces a new feature
  - breaking       # the API has changed in a breaking way
  # - compatible     # the API has changed but is non-breaking
  # - optimisation   # measurable performance improvements
  - improvement    # QoL changes e.g. refactoring
  # - bugfix         # fixes a defect
  # - test           # fixes/modifies tests
  # - maintenance    # not directly related to the code
  # - release        # related to a new release preparation
  # - documentation  # change in code docs, haddocks...
```

# Context

Closes https://github.com/input-output-hk/cardano-cli/issues/198

# Checklist

- [X] Commit sequence broadly makes sense and commits have useful messages
- [X] The change log section in the PR description has been filled in
- [ ] New tests are added if needed and existing tests are updated.  These may include:
  - golden tests
  - property tests
  - round trip tests
  - integration tests
  See [Running tests](https://github.com/input-output-hk/cardano-node-wiki/wiki/Running-tests) for more details
- NA The version bounds in `.cabal` files are updated
- [X] CI passes. See note on CI.  The following CI checks are required:
  - [X] Code is linted with `hlint`.  See `.github/workflows/check-hlint.yml` to get the `hlint` version
  - [X] Code is formatted with `stylish-haskell`.  See `.github/workflows/stylish-haskell.yml` to get the `stylish-haskell` version
  - [X] Code builds on Linux, MacOS and Windows for `ghc-8.10.7` and `ghc-9.2.7`
- [X] The changelog section in the PR is updated to describe the change
- [X] Self-reviewed the diff

### CODE REVIEW

**Review by CarlosLopezDeLara — APPROVED** [review]
Looks good! thank you

```
➜  testdreps cardano-cli conway governance drep registration-certificate \
--drep-verification-key-file drep.vkey \
--key-reg-deposit-amt 0 \
--drep-metadata-url drep.url \
--drep-metadata-hash 52e69500a92d80f2126c836a4903dc582006709f004cf7a28ed648f732dff8d2 \
--out-file drep-reg.cert
➜  testdreps cat drep-reg.cert
{
    "type": "CertificateShelley",
    "description": "DRep Key Registration Certificate",
    "cborHex": "84108200581c95591451fb3899fcc8db3450741d9de6123a78e03ef39bf4a65ed5cf008268647265702e75726c582052e69500a92d80f2126c836a4903dc582006709f004cf7a28ed648f732dff8d2"
}
```

`[16, [0, h'95591451FB3899FCC8DB3450741D9DE6123A78E03EF39BF4A65ED5CF'], 0, ["drep.url", h'52E69500A92D80F2126C836A4903DC582006709F004CF7A28ED648F732DFF8D2']]`

**Comment by smelc** [comment]
cc @Jimbo4350 and @newhoggy if you want to have a look before merging. Short simple PR with green CI.

**Comment by Jimbo4350** [comment]
Needs to be rebased. Having a look now.

File: cardano-cli/src/Cardano/CLI/EraBased/Run/Governance/DRep.hs [hunk 1]
```diff
@@ -69,21 +71,23 @@ runGovernanceDRepIdCmd _ vkOrFp idOutputFormat mOutFile = do
 -- Registration Certificate related
 
 runGovernanceRegistrationCertificateCmd
-  :: ConwayEraOnwards era
+  :: forall era ledgerera. ()
```
Comment (Jimbo4350) [inline cardano-cli/src/Cardano/CLI/EraBased/Run/Governance/DRep.hs]: So we can simplify this type signature as follows:
```
runGovernanceRegistrationCertificateCmd
  :: ConwayEraOnwards era
  -> VerificationKeyOrHashOrFile DRepKey
  -> Lovelace
  -> Maybe (Ledger.Anchor (Ledger.EraCrypto (ShelleyLedgerEra era)))
  -> File () Out
  -> ExceptT RegistrationError IO ()
runGovernanceRegistrationCertificateCmd ...
```
We should also be able to remove the `TypeOperators` extension.

File: cardano-cli/src/Cardano/CLI/Types/Common.hs [hunk 2]
```diff
@@ -127,6 +129,14 @@ data ProposalHashSource
   | ProposalHashSourceHash (L.SafeHash Crypto.StandardCrypto L.AnchorData)
   deriving Show
 
+newtype AnchorUrl = AnchorUrl
```
Comment (Jimbo4350) [inline cardano-cli/src/Cardano/CLI/Types/Common.hs:132]: This is fine here for now but we should move them to `cardano-api` in a follow up PR.

**Review by Jimbo4350 — APPROVED** [review]
LGTM! One simplifying suggestion

File: cardano-cli/src/Cardano/CLI/Types/Common.hs (hunk 2 above)
Comment (smelc) [reply cardano-cli/src/Cardano/CLI/Types/Common.hs:132]: Sounds good :+1: Created follow-up issue to keep track: https://github.com/input-output-hk/cardano-api/issues/301

File: cardano-cli/src/Cardano/CLI/EraBased/Run/Governance/DRep.hs (hunk 1 above)
Comment (smelc) [reply cardano-cli/src/Cardano/CLI/EraBased/Run/Governance/DRep.hs]: Simpler indeed, I did the change :+1:

**Pull request merged**